### DISCUSSION TITLE
Web "Playground"

### DISCUSSION THREAD

**Discussion by k9withabone** [k9withabone/podlet#20] [Ideas]
I'm thinking of creating a podlet web "playground", similar to playgrounds for programming languages, like the [rust playground](https://play.rust-lang.org). You would input your podman command or compose file and see the generated quadlet.
The first step is to move the conversion code into a separate library that I can use via WASM. Then I can use something like [leptos](https://github.com/leptos-rs/leptos) to create the page. It can then be hosted via GitHub pages since it will be fully static.

**Comment by mikkovihonen** [comment]
@k9withabone I made a web ui for slightly different purposes but what do you think, would podlet integration make sense here: [https://github.com/mikkovihonen/quadletman](https://github.com/mikkovihonen/quadletman)

**Reply by k9withabone** [reply]
Looks like an interesting project! Once I publish Podlet as a library you could possibly use [PyO3](https://github.com/PyO3/pyo3) to create Python bindings. Out of curiosity, why did you go with a separate application as opposed to contributing to cockpit-podman or making your own cockpit plugin?
